pkg/whitelist/whitelist.go

```diff
@@ -0,0 +1,81 @@
+package whitelist
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ghodss/yaml"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	cioperatorapi "github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/promotion"
+)
+
+// WhitelistConfig holds a list of repositories mapped by organization
+type WhitelistConfig struct {
+	Whitelist map[string][]string `json:"whitelist,omitempty"`
+}
+
+func (w *WhitelistConfig) IsWhitelisted(info *config.Info) bool {
+	if whiteRepos, ok := w.Whitelist[info.Org]; ok {
+		if sets.NewString(whiteRepos...).Has(info.Repo) {
+			return true
+		}
+	}
+	return false
+}
+
+// Options holds the required flags to load the whitelist configuration
+type Options struct {
+	whitelistFile string
+}
+
+// Validate validates if the whitelist cofiguration file actual exists.
+func (o *Options) Validate() error {
+	if o.whitelistFile != "" {
+		info, err := os.Stat(o.whitelistFile)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("The file that specified in --whitelist-file does not exist: %v", o.whitelistFile)
+		}
+
+		if info.IsDir() {
+			return fmt.Errorf("The file that specified in --whitelist-file is a directory: %v", o.whitelistFile)
+		}
+	}
+	return nil
+}
+
+func (o *Options) Bind(fs *flag.FlagSet) {
+	fs.StringVar(&o.whitelistFile, "whitelist-file", "", "File of the repository whitelist configuration")
+}
+
+// LoadWhitelist reads the whitelist configuration file and returns a WhitelistConfig object.
+func (o *Options) LoadWhitelist() (WhitelistConfig, error) {
+	var whitelistConfig WhitelistConfig
+	if o.whitelistFile != "" {
+		bytes, err := ioutil.ReadFile(o.whitelistFile)
+		if err != nil {
+			return WhitelistConfig{}, fmt.Errorf("Couldn't read whitelist configuration file: %v", o.whitelistFile)
+		}
+		if err := yaml.Unmarshal(bytes, &whitelistConfig); err != nil {
+			return WhitelistConfig{}, errors.New("Couldn't unmarshal whitelist configuration")
+		}
+	}
+	return whitelistConfig, nil
+}
+
+// OperateOnCIOperatorConfigDir filters the full set of configurations
+// down to those that exist in the whitelist configuration and build official images
+func OperateOnCIOperatorConfigDir(configDir string, callback func(*cioperatorapi.ReleaseBuildConfiguration, *config.Info) error, whitelistConfig WhitelistConfig) error {
```

Comment:
This should be a method on `(o *Options)` -- if someone is using a tool with a whitelist, it should be impossible for them to forget to add it.